castervoice/apps/ide/ide.py

```diff
@@ -0,0 +1,94 @@
+#
+# __author__ = "lexxish"
+#
+from castervoice.apps.shared.directions import FORWARD, RIGHT, BACK, LEFT, UP, DOWN
+
+method = "(meth|method)"
+
+# general
+EXPAND_SELECTION = "expand [selection] [<n>]"
+SHRINK_SELECTION = "shrink [selection] [<n>]"
+SMART_AUTO_COMPLETE = "(skraken|smart kraken)"
+GENERATE_CODE = "jen code"
+QUICK_FIX = "quickfix"
+FORMAT_ALL_CODE = "format [code]"
+BUILD_PROJECT = "build"
+RUN_PROJECT = "run"
+DEBUG_PROJECT = "debug"
+BUILD_AND_RUN_PROJECT = "build and run"
+DEBUG_CURRENT_FILE = "debug file"
+RUN_CURRENT_FILE = "run file"
+NEXT_ERROR = "(%s error|error %s)" % (FORWARD, RIGHT)
+PREVIOUS_ERROR = "(%s error|error %s)" % (BACK, LEFT)
+REDO = "redo [<n>]"
+SHOW_SETTINGS = "[show] settings"
+RENAME_CURRENT_FILE = "file rename | rename file"
+
+# window navigation
+NEXT_TAB = "%s tab [<n>]|tab %s [<n>]" % (FORWARD, RIGHT)
+PREVIOUS_TAB = "%s tab [<n>]|tab %s [<n>]" % (BACK, LEFT)
+CLOSE_TAB_N_TIMES = "close tab [<n>]|tab close [<n>]"
+GO_TO_EDITOR = "go [to] editor"
+GO_TO_PROJECT_EXPLORER = "go [to] project"
+TOGGLE_TERMINAL = "[toggle] (term|terminal)"
+NEW_FILE = "new file"
+
+# editor management
+SPLIT_WINDOW_UP = "split [pane] %s" % UP
+SPLIT_WINDOW_DOWN = "split [pane] %s" % DOWN
+SPLIT_WINDOW_RIGHT = "split [pane] %s" % RIGHT
+SPLIT_WINDOW_LEFT = "split [pane] %s" % LEFT
+SPLIT_MOVE_UP = "pane %s [<n>]" % UP
+SPLIT_MOVE_DOWN = "pane %s [<n>]" % DOWN
+SPLIT_MOVE_RIGHT = "(pane %s|next pane) [<n>]" % RIGHT
+SPLIT_MOVE_LEFT = "(pane %s|(prior|previous) pane) [<n>]" % LEFT
```

Comment:
why not have all the BACK's here too (why only prior|previous?).

Reply:
was intentional (thought they didn't sound right), but changed it